### PR TITLE
Compatibility with PHP 8.1

### LINKED ISSUE #68: PHP8.1: imagefilledpolygon(): Using the $num_points parameter is deprecated
Just an early heads-up:
Using the PHP8.1 beta2 the following deprecation warning is shown:
> Deprecated - imagefilledpolygon(): Using the $num_points parameter is deprecated

https://github.com/szymach/c-pchart/blob/c71e1d8f08037148e684dd7b7bf22a3d2eeefe06/src/Draw.php#L81-L83

see also https://www.php.net/manual/en/function.imagefilledpolygon.php

### LINKED ISSUE #69: PHP8.1: Implicit conversion from float to int loses precision
Another early heads-up:

> Deprecated - Implicit conversion from float 74.08333333332371 to int loses precision

https://github.com/szymach/c-pchart/blob/c71e1d8f08037148e684dd7b7bf22a3d2eeefe06/src/BaseDraw.php#L288
https://github.com/szymach/c-pchart/blob/c71e1d8f08037148e684dd7b7bf22a3d2eeefe06/src/BaseDraw.php#L510
https://github.com/szymach/c-pchart/blob/c71e1d8f08037148e684dd7b7bf22a3d2eeefe06/src/Draw.php#L1312
see https://wiki.php.net/rfc/implicit-float-int-deprecate

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #69
fixes #68

### CODE REVIEW

**Comment by szymach** [comment]
Hello, could you please change the base of the PR to the `3.0` branch? The `master` branch is currently in the process of BC breaking changes.

Thank you.

**Comment by sgiehl** [comment]
@szymach done

**Comment by szymach** [comment]
Hello, 
unfortunately this is a BC break, as you can see from the failures in the test suite. Could you please make a wrapper function for it, which would utilize the [version_compare](https://www.php.net/manual/en/function.version-compare) function? Something like:

```php
/**
 * @param GdImage|resource $image
 * @param array $points
 * @param int $numPoints
 * @param int $color
 * @return void
 */
protected function imageFilledPolygonWrapper(
    $image,
    array $points,
    $numPoints,
    $color
)  {
    if (version_compare(PHP_VERSION, '8.1.0') === -1) {
        imagefilledpolygon($image, $points, $numPoints, $color); 
    } else {
        imagefilledpolygon($image, $points, $color); 
    }
}
```
It should go in the `BaseDraw` class. Thank you.

**Comment by sgiehl** [comment]
@szymach done. Seems tests are passing again :-)

File: src/BaseDraw.php [hunk 1]
```diff
@@ -516,7 +516,7 @@ public function modulo($Value1, $Value2)
             $Factor = $Factor * 10;
         }
 
-        return ($Value1 * $Factor) % ($Value2 * $Factor);
+        return floor($Value1 * $Factor) % floor($Value2 * $Factor);
```
Comment (szymach) [inline src/BaseDraw.php:519]: Can you please tell me how do the values here change after adding these?

Comment (sgiehl) [reply src/BaseDraw.php:519]: They actually shouldn't change at all. The floor is, what PHP will do in background. The modulo operation (`%`) works with integers only. And when casting a float to integer PHP simply cuts off the fraction digits.
See this for a simple demonstration: https://3v4l.org/EehJN

**Review by szymach — APPROVED** [review]
Thank you for the PR.

**Pull request merged**